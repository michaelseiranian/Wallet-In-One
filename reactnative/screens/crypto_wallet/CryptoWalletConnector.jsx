import React, { useState } from "react";
import {
  ActivityIndicator,
  Button, Image,
  StyleSheet,
  Text,
  TextInput,
  View,
} from "react-native";
import getCryptoIcon from "../cryptocurrency/icons/icon";
import { useTheme } from '../../src/theme/ThemeProvider';
import {styles} from '../All_Styles.style.js';


/**
 * Component that allows a user to connect a crypto wallet by entering the address into the corresponding cryptocurrency
 * connector screen.
 */
export default function CryptoWalletConnector(props) {
  const [address, setAddress] = useState("");
  const [loading, setLoading] = useState(false);
  const { connectWallet, cryptocurrency, symbol } = props.route.params;
  const {dark, colors, setScheme} = useTheme();

  const stylesInternal = StyleSheet.create({
    input: {
      height: 40,
      width: '100%',
      borderWidth: 0.5,
      padding: 10,
      borderColor: 'gray',
      borderRadius: 5,
      marginTop: 5,
      marginBottom: 5,
    },
  });

  return (
    <View style={{flex:1, backgroundColor: colors.background, paddingHorizontal: 20}}>
      <View style={{ paddingTop: 30 }}>
        <View style={{ flexDirection: "row", alignItems: "center" }}>
          <View style={{ flex: 1, flexDirection: "column" }}>
            <Text style={[styles(dark, colors).largeTextBold, {alignSelf: "center"}]}>Connect Wallet</Text>
          </View>
        </View>

        <View style={{alignItems: "center", justifyContent: "center", paddingTop: 30}}>
          <Image
            style={{width: 100, height: 100}}
            source={getCryptoIcon(symbol)}
          />
          <Text style={{fontWeight: "800", fontSize: 30, alignSelf: "center", color: colors.text}}>{cryptocurrency}</Text>
        </View>

        <TextInput
          autoCapitalize='none'
          autoCorrect={false}
          style={[styles(dark, colors).input, {color: colors.text}, {backgroundColor: colors.background}]}
          onChangeText={(text) => setAddress(text)}
          placeholderTextColor= {colors.text}
          placeholder="Wallet Address"
          testID="addressInput"
        />

        {
          loading ?
            <ActivityIndicator size="large" color={colors.primary}/>
            :
            <Button
              title="Connect Wallet"
              onPress={() => {
                  setLoading(true);
                  connectWallet(cryptocurrency, symbol, address)
                    .then(() => props.navigation.navigate("Crypto Wallets & Exchanges"))
                }
              }
            />
        }
      </View>
    </View>
  );
}
