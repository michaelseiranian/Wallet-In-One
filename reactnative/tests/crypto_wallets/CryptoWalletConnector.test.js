import {act, fireEvent, render, screen, waitFor} from "@testing-library/react-native";
import CryptoWalletConnector from "../../screens/crypto_wallet/CryptoWalletConnector";

describe('<CryptoWalletConnector />', () => {

  const params = {
    route: {
      params: {
          currentBid: 'whatever-id',
          connectWallet: jest.fn(async () => {return Promise}),
        },
    },
    navigation: {
      navigate: jest.fn(),
    }
  };

  it('snapshot test', async () => {
    const snapshot = render(<CryptoWalletConnector {...params} />)

    await act(async () => {
      expect(snapshot).toMatchSnapshot()

      // Test navigation
      const connectButton = screen.getAllByText('Connect Wallet')[1]
      fireEvent.press(connectButton)

    })
  })

  it('address change test', async () => {
    const testValue = "Test"

    const { getByTestId } = render(<CryptoWalletConnector {...params} />)
    const textInput = getByTestId('addressInput')
    fireEvent.changeText(textInput, testValue)

  })

})