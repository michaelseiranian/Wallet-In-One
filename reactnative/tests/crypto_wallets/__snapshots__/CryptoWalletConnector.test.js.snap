// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoWalletConnector /> snapshot test 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "black",
                "fontSize": 40,
                "fontWeight": "800",
              },
              {
                "alignSelf": "center",
              },
            ]
          }
        >
          Connect Wallet
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingTop": 30,
        }
      }
    >
      <Image
        style={
          {
            "height": 100,
            "width": 100,
          }
        }
      />
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "black",
            "fontSize": 30,
            "fontWeight": "800",
          }
        }
      />
    </View>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Wallet Address"
      placeholderTextColor="black"
      style={
        [
          {
            "borderColor": "gray",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "height": 40,
            "marginBottom": 5,
            "marginTop": 5,
            "padding": 10,
            "width": "100%",
          },
          {
            "color": "black",
          },
          {
            "backgroundColor": "white",
          },
        ]
      }
      testID="addressInput"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Connect Wallet
        </Text>
      </View>
    </View>
  </View>
</View>
`;
